app/src/main/java/org/simple/clinic/registration/facility/RegistrationFacilitySelectionEffectHandler.kt

```diff
@@ -70,4 +77,16 @@ class RegistrationFacilitySelectionEffectHandler @AssistedInject constructor(
           }
     }
   }
+
+  private fun saveCurrentRegistrationEntry(): ObservableTransformer<SaveRegistrationEntryAsUser, RegistrationFacilitySelectionEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .doOnNext { userSession.saveOngoingRegistrationEntry(it.entry) }
```

Comment:
Should we these operations on `io` thread?

Reply:
We don't need to since they do not do any I/O at all and are just objects in memory. As a conclusion of this migration, I have completely removed user session for being responsible for storing this ongoing entry and made it an object that is passed via the screen keys. This will be seen in later PRs.